core/server/data/migrations/versions/4.3/04-attach-members-to-product.js

```diff
@@ -0,0 +1,48 @@
+const {createTransactionalMigration} = require('../../utils');
+const ObjectID = require('bson-objectid');
+const {chunk} = require('lodash');
+const logging = require('../../../../../shared/logging');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const membersWithProduct = await knex
+            .select('id')
+            .from('members')
+            .whereIn('status', ['comped', 'paid']);
+
+        if (membersWithProduct.length === 0) {
+            logging.info(`No members found with product`);
+            return;
+        }
+
+        const product = await knex
+            .select('id', 'name')
+            .from('products')
+            .first();
+
+        if (!product) {
+            logging.warn(`No product found to attach members to`);
+            return;
+        }
+
+        logging.info(`Attaching product ${product.name} to ${membersWithProduct.length} members`);
+        const memberProductRelations = membersWithProduct.map((member) => {
+            return {
+                id: ObjectID.generate(),
+                member_id: member.id,
+                product_id: product.id
+            };
+        });
+
+        const chunkSize = 333;
```

Comment:
Not blocking: it could be nice to add the reason why it's 333 in a comment (I think I've seen such a comment in a few other migrations).

Reply:
Done :)